src/main/java/de/triology/cas/services/RegistryEtcd.java

```diff
@@ -77,6 +78,59 @@ public String getFqdn() {
         }
     }
 
+    public URI getCasLogoutUri(String doguname) throws GetCasLogoutUriException {
+        JSONObject doguMetaData;
+        try {
+            EtcdKeysResponse.EtcdNode node = getDoguNodeFromEtcd(doguname);
+            doguMetaData = getCurrentDoguNode(node);
+            JSONObject properties;
+            if (doguMetaData != null) {
+                properties = getPropertiesFromMetaData(doguMetaData);
+            } else {
+                throw new GetCasLogoutUriException("Could not get dogu metadata");
+            }
+            return getLogoutUriFromProperties(properties);
+        } catch (ParseException | URISyntaxException | GetDoguNodeFromEtcdException e) {
+            throw new GetCasLogoutUriException(e.toString());
+        }
+    }
+
+    private URI getLogoutUriFromProperties(JSONObject properties) throws GetCasLogoutUriException, URISyntaxException {
+        Object logoutUri = properties.get("logoutUri");
+        if (logoutUri != null){
+            String logoutUriString = logoutUri.toString();
+            if (logoutUriString != null) {
+                return new URI(logoutUriString);
+            } else {
+                throw new GetCasLogoutUriException("Could not get logoutUri from properties");
+            }
+        } else {
+            throw new GetCasLogoutUriException("Could not get logoutUri from properties");
+        }
+    }
+
+    private JSONObject getPropertiesFromMetaData(JSONObject doguMetaData) throws GetCasLogoutUriException {
```

Comment:
Is this a GetCasLogoutUriException?
We could avoid this Problem if we would use the generic 'RegistryException' like the other Methods in this Class.